code/createDeviceSettingsTable.m

```diff
@@ -0,0 +1,757 @@
+            
+        end
+        
+        % FFT
+        if inStream_FFT && ~currentSettings.StreamState.FftStreamEnabled
+            HostUnixTime = currentSettings.RecordInfo.HostUnixTime;
+            FFT_SettingsTable.action{entryNumber_FFT} = sprintf('Stop Stream FFT %d',streamStopCounter_FFT);
+            FFT_SettingsTable.recNum(entryNumber_FFT) = streamStopCounter_FFT;
+            FFT_SettingsTable.time{entryNumber_FFT} = HostUnixTime;
+            
+            % Fill in most recent FFT settings
+            FFT_SettingsTable.fftConfig(entryNumber_FFT) = fftConfig;
+            
+            % Get sample rate for each TD channel; all TD channels have
+            % same Fs (or is listed as NaN)
+            for iChan = 1:4
+                TDsampleRates(iChan) = str2double(TDsettings(iChan).sampleRate(1:end-2));
+            end
+            TDsampleRates = unique(TDsampleRates);
+            currentTDsampleRate = TDsampleRates(~isnan(TDsampleRates));
+            FFT_SettingsTable.TDsampleRates{entryNumber_FFT} = currentTDsampleRate;
+            
+            inStream_FFT = 0;
+            streamStopCounter_FFT = streamStopCounter_FFT + 1;
+            entryNumber_FFT = entryNumber_FFT + 1;
+        end
+    end
+    
+    % Option 2: Check if sense has been turned off
+    if isfield(currentSettings,'SenseState')
+        % TIME DOMAIN
+        if inStream_TD && isfield(currentSettings.SenseState,'state')
+            senseState = dec2bin(currentSettings.SenseState.state,4);
+            % Check starting/stopping of time domain streaming. See
+            % documentation enum Medtronic.NeuroStim.Olympus.DataTypes.Sensing.SenseStates : byte
+            % for more details about binary number coding
+            
+            if strcmp(senseState(4),'0') % Time domain streaming is off
+                % Create new row in deviceSettingsTable and populate with metadata
+                HostUnixTime = currentSettings.RecordInfo.HostUnixTime;
+                TD_SettingsTable.action{entryNumber_TD} = sprintf('Stop Sense TD %d',streamStopCounter_TD);
+                TD_SettingsTable.recNum(entryNumber_TD) = streamStopCounter_TD;
+                TD_SettingsTable.time{entryNumber_TD} = HostUnixTime;
+                
+                % Fill in most recent time domain data settings
+                for iChan = 1:4
+                    fieldName = sprintf('chan%d',iChan);
+                    TD_SettingsTable.(fieldName){entryNumber_TD} = TDsettings(iChan).chanFullStr;
+                end
+                TD_SettingsTable.tdDataStruc{entryNumber_TD} = TDsettings;
+                inStream_TD = 0;
+                streamStopCounter_TD = streamStopCounter_TD + 1;
+                entryNumber_TD = entryNumber_TD + 1;
+            end
+        end
+        
+        % POWER DOMAIN
+        if inStream_Power && isfield(currentSettings.SenseState,'state')
+            senseState = dec2bin(currentSettings.SenseState.state,4);
```

Comment:
Redundant with time domain check. Look at suggestion above.

Reply:
```suggestion

```

Reply:
you can only stream power and not time domain - not sure what is redundant ?

Reply:
Specifically the conversion from decimal to binary 
`senseState = dec2bin(currentSettings.SenseState.state,4);`
In option 2 it is calculates this 3 times, once in TD, Power, and FFT.

Reply:
Moved the check of senseState to outside the loops dependent on whether streaming for TD, Power, or FFT. Updated with f6e367ec1556dac0b4ab4ec47597070379c6d8f2